Adding a system of recommendation

m-rohmer opened the pull request:
Modifiying swagger.yaml, the product's model files and the test part

m-rohmer committed:
Adding a system of recommendation

Modifiying swagger.yaml, the product's model files and the part test

rouvoy commented on model/product/product-facade.js:
```diff
@@ -126,6 +127,60 @@ class ProductModel extends Model {
 
   }
 
+    findProductsByAssociation(eansProducts) {
```
Incorrect indentation for this method

rouvoy commented on test/product-test.js:
```diff
@@ -311,10 +311,10 @@ describe('Products', () => {
-      .set('Accept', 'application/json')
-      .expect(200)
-      .end((err, res) => {
+	    .expect('Content-type', /json/)
```
Incorrect indentation?

rouvoy commented on model/product/product-controller.js:
```diff
@@ -197,6 +232,47 @@ class ProductController extends Controller {
+	    .findProductsByAssociation(req.query.eans)
+	    .then((collection) => {
+		if (collection === null || collection.length === 0) {
+		    res.status(404).send('No product associated with those products');
```
Not sure we should return a `404` error in such a case, as there is nothing wrong with the incoming request sent by the app. There is just no associated product so far, but we cannot blame the app for that.

rouvoy commented on model/product/product-controller.js:
```diff
@@ -70,6 +70,41 @@ function checkArrayBeacon(query) {
+
 
 function checkParam(req, params, eanForced = false) {
   // Test for invalid params
```
Missing body?

rouvoy commented on model/product/product-controller.js:
```diff
@@ -58,7 +58,7 @@ function checkArrayBeacon(query) {
 
   if (!query){
-    return { message: `Param doesn't exists`, code: 400 };
+    return { message: `Param doesn\'t exists`, code: 400 };
```
Replace by `Unknown parameter `+ query

rouvoy commented on dist/swagger.yaml:
```diff
@@ -109,6 +110,62 @@ paths:
+              $ref: '#/definitions/Product'
+      responses:
+        '200':
+          description: Excected response to a valid request
```
Expected

m-rohmer committed:
Changing return when no association found

Changing response when request to find a product by beacons or other product association doesn't match

m-rohmer committed:
Using js-beautify to correct indentation
